modin/pandas/test/test_series.py

```diff
@@ -3135,7 +3135,7 @@ def test_sort_values(data, ascending, na_position):
     # between `pandas.Series.sort_values`. For this reason, we check that the values are
     # identical instead of the index as well.
     if ascending:
-        df_equals(modin_result, pandas_result)
+        df_equals_with_non_stable_indices(modin_result, pandas_result)
```

Comment:
I had to make this change because the data contains rows with duplicate values, and so the order of the indexes may be different. Should we just get rid of the if-else and just do `np.testing.assert_equal`?

Reply:
I wonder if we can do this _only_ for the cases where data has duplicates... or are you saying that all our data options have duplicates?

also I wonder if we can make sorting stable... or is it already?

Reply:
Our data is generated randomly, so all of our data can have duplicates. I don't think sort is stable since it depends on random sampling.

Reply:
Do wa have a warning or note for this?

Reply:
@anmyachev has a PR for #5364

Reply:
Not sure if it is related. AFAIU, with the new `sort_values` implementation the order of the indexes may be different. I wonder if we should warn users about that or add a note in this page - https://modin.readthedocs.io/en/stable/supported_apis/dataframe_supported.html?

Reply:
We could add a note that says that sort is not stable for different indices with the same key value?

Reply:
SGTM!